File: app/src/main/java/com/magdenbt/collectionsbenchmark/ui/viewmodelflow/StatDiffCallback.java
package com.magdenbt.collectionsbenchmark.ui.viewmodelflow;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.recyclerview.widget.DiffUtil;

import com.magdenbt.collectionsbenchmark.modelflow.StatModel;


public class StatDiffCallback extends DiffUtil.ItemCallback<LiveData<StatModel>> {

    public StatDiffCallback() {
    }

    @Override
    public boolean areItemsTheSame(@NonNull LiveData<StatModel> oldItem, @NonNull LiveData<StatModel> newItem) {
        return oldItem.getValue().equals(newItem.getValue());
    }

    @Override
    public boolean areContentsTheSame(@NonNull  LiveData<StatModel> oldItem, @NonNull  LiveData<StatModel> newItem) {
        return oldItem.getValue().getBusy() == newItem.getValue().getBusy() && oldItem.getValue().getStatus().equals(newItem.getValue().getStatus()) ;
    }

}

File: app/src/main/java/com/magdenbt/collectionsbenchmark/ui/viewmodelflow/StatDiffCallback.kt
package com.magdenbt.collectionsbenchmark.ui.viewmodelflow

import androidx.lifecycle.LiveData
import androidx.recyclerview.widget.DiffUtil
import com.magdenbt.collectionsbenchmark.modelflow.StatModel

class StatDiffCallback : DiffUtil.ItemCallback<LiveData<StatModel>>() {

    override fun areItemsTheSame(
        oldItem: LiveData<StatModel>, newItem: LiveData<StatModel>
    ) = oldItem.value == newItem.value

    override fun areContentsTheSame(
        oldItem: LiveData<StatModel>, newItem: LiveData<StatModel>
    ) =
        oldItem.value!!.busy == newItem.value!!.busy && oldItem.value!!.status == newItem.value!!.status
}